students/roviram/Lesson02/FibonacciSeries.py

```diff
@@ -0,0 +1,80 @@
+#-----------------------------------------------------------#
+# Dev: Miguel Rovira-Gonzalez
+# Desc: Fibnonacci Series Homework, Week 2
+# Date Created: Sunday 4/21
+#-----------------------------------------------------------#
+def fibonacci(n):
+    """"
+    Compute the nth Fibnonacci number
+    fib(n) = fib(n-2) + fib(n-1)
+    """
+
+    a = 0
+    b = 1
+    for number in range(n):
+        a, b = b, a + b
+    return a
+
+def lucas(n):
+    """ compute the nth Lucas number, values start with 2 and 1 rather than 0
+    2, 1, 3, 4, 7
+    """
+    a = 2
+    b = 1
+    for number in range(n):
+        a, b = b, a + b
+    return a
+
+def sum_series(n, n0=0, n1=1):
+    """
+    compute the nth value of a summation series.
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+    This function should generalize the fibonacci() and the lucas(),
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+
+    sum_series(n, 3, 2) should generate another series with no specific name
+
+    The defaults are set to 0, 1, so if you don't pass in any values, you'll
+    get the fibonacci series
+    """
+    for number in range(n):
+        n0, n1 = n1, n0 + n1
+    return n0
```

Comment:
it's actually pretty slick how computing the extra value means you don't need to special-case n=0 or n=1.

Did you do that on purpose?